tools/monitoring/mysql/error_logger.go

```diff
@@ -18,50 +18,38 @@ package mysql
 
 import (
 	"database/sql"
+	"fmt"
 	"time"
 
 	"github.com/knative/test-infra/shared/mysql"
-	"github.com/knative/test-infra/tools/monitoring/config"
-	"github.com/knative/test-infra/tools/monitoring/log_parser"
 )
 
-const (
-	logInsertStmt = `
-	INSERT INTO ErrorLogs (
-		ErrorPattern, ErrorMsg, JobName, PRNumber, BuildLogURL, TimeStamp
-		) VALUES (?,?,?,?,?,?)`
-)
+// MonitoringDatabase holds an active database connection created in `config`
+type MonitoringDatabase struct {
+	*sql.DB
+	config mysql.DBConfig
+}
 
-// PubsubMsgHandler adds record(s) to ErrorLogs table in database,
-// after parsing build log and compares the result with config yaml
-func PubsubMsgHandler(db *sql.DB, configURL, buildLogURL, jobname string, prNumber int) error {
-	config, err := config.ParseYaml(configURL)
-	if err != nil {
-		return err
-	}
+func (db *MonitoringDatabase) insertErrorLog(errPat string, errMsg string, jobName string, prNum int, blogURL string) error {
+	stmt, err := db.Prepare(`INSERT INTO ErrorLogs(ErrorPattern, ErrorMsg, JobName, PRNumber, BuildLogURL, TimeStamp)
+				VALUES (?, ?, ?, ?, ?, ?)`)
+	defer stmt.Close()
 
-	errorLogs, err := log_parser.ParseLog(buildLogURL, config.CollectErrorPatterns())
-	if err != nil {
-		return err
-	}
+	_, err = execAffectingOneRow(stmt, errPat, errMsg, jobName, prNum, blogURL, time.Now())
+	return err
+}
 
-	tx, err := db.Begin()
+// execAffectingOneRow executes a given statement, expecting one row to be affected.
```

Comment:
Why do we need this? A successful insert statement will only affect one row. I think this is redundant

Reply:
It is true that `insert` will only affect one row. But the type of sql statements to execute by this function is not limited to insert. For example, this check also checks that update statements only affects one row.

Reply:
Yes, but that is not reflected in this PR. I would prefer to see it used in more than one places to justify the refactoring. Also, I feel the name is too long and does not look go-ish.

Reply:
Removed